pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/PredicateComparisonRewriter.java

```diff
@@ -33,75 +36,126 @@ public class PredicateComparisonRewriter implements QueryRewriter {
+            // Handle predicate like '10 = a' -> 'a = 10'
+            if (firstOperand.isSetLiteral()) {
+              if (!secondOperand.isSetLiteral()) {
+                function.setOperator(getOppositeOperator(filterKind).name());
+                operands.set(0, secondOperand);
+                operands.set(1, firstOperand);
+              }
+              break;
+            }
+
+            // Handle predicate like 'a > b' -> 'a - b > 0'
+            if (!secondOperand.isSetLiteral()) {
+              Expression minusExpression = RequestUtils.getFunctionExpression("minus");
+              minusExpression.getFunctionCall().setOperands(Arrays.asList(firstOperand, secondOperand));
+              operands.set(0, minusExpression);
+              operands.set(1, RequestUtils.getLiteralExpression(0));
+              break;
+            }
             break;
-          }
-          break;
-        default:
-          int numOperands = operands.size();
-          for (int i = 1; i < numOperands; i++) {
-            if (!operands.get(i).isSetLiteral()) {
-              throw new SqlCompilationException(
-                  String.format("For %s predicate, the operands except for the first one must be literal, got: %s",
-                      filterKind, expression));
+          default:
+            int numOperands = operands.size();
+            for (int i = 1; i < numOperands; i++) {
+              if (!operands.get(i).isSetLiteral()) {
+                throw new SqlCompilationException(String
+                    .format("For %s predicate, the operands except for the first one must be literal, got: %s",
+                        filterKind, expression));
+              }
             }
-          }
-          break;
+            break;
+        }
       }
+    } else if (type == ExpressionType.IDENTIFIER) {
+      expression = convertPredicateToBooleanExpression(expression, parentFunction);
     }
+
+    return expression;
+  }
+
+  /**
+   * Rewrite if one of the two conditions are satisfied:
+   *  1. parent function is empty.
+   *     Example1: "select * from table where col1" converts to
+   *                "select * from table where col1 = true"
+   *     Example2: "select * from table where startsWith(col1, 'str')" converts to
+   *               "select * from table where startsWith(col1, 'str') = true"
+   *  2. Separator is not a predicate (non predicates are AND, OR, NOT).
+   *     Example1: "select * from table where col1 AND startsWith(col2, 'str')" converts to
+   *               "select * from table where col1 = true AND startsWith(col2, 'str') = true"
+   * @param expression Expression
+   * @param parentFunction Parent expression
+   * @return Rewritten expression
+   */
+  private static Expression convertPredicateToBooleanExpression(Expression expression, Function parentFunction) {
+    if (parentFunction == null || (EnumUtils.isValidEnum(FilterKind.class, parentFunction.getOperator()) && !FilterKind
```

Comment:
This condition should always be met, and we should be able to rewrite the expression without any check. `parentFunction` is not needed here

Reply:
I see your point now. The code is much cleaner following this suggestion. Thanks!